packages/flutter/lib/src/widgets/widget_span.dart

```diff
@@ -0,0 +1,262 @@
+  /// decoration, foreground, background, and spacing options will be used.
+  const WidgetSpan({
+    Widget widget,
+    InlineWidgetAlignment alignment = InlineWidgetAlignment.bottom,
+    TextBaseline baseline,
+    TextStyle style,
+    List<InlineSpan> children,
+  }) : assert(widget != null),
+       assert((alignment == InlineWidgetAlignment.aboveBaseline ||
+               alignment == InlineWidgetAlignment.belowBaseline ||
+               alignment == InlineWidgetAlignment.baseline) ? baseline != null : true),
+       widget = widget,
+       super(alignment:
+           // Convert InlineWidgetAlignment to PlaceholderAlignment in a const fashion.
+           alignment == InlineWidgetAlignment.baseline ? ui.PlaceholderAlignment.baseline :
+           alignment == InlineWidgetAlignment.aboveBaseline ? ui.PlaceholderAlignment.aboveBaseline :
+           alignment == InlineWidgetAlignment.belowBaseline ? ui.PlaceholderAlignment.belowBaseline :
+           alignment == InlineWidgetAlignment.top ? ui.PlaceholderAlignment.top :
+           alignment == InlineWidgetAlignment.bottom ? ui.PlaceholderAlignment.bottom :
+           alignment == InlineWidgetAlignment.middle ? ui.PlaceholderAlignment.middle : null,
+         baseline: baseline, style: style, children: children,
+       );
+
+  /// The widget to embed inline with text.
+  final Widget widget;
+
+  /// Adds a placeholder box to the paragraph builder if a size has been
+  /// calculated for the widget.
+  ///
+  /// Sizes are provided through [dimensions], which should contain a 1:1
+  /// in-order mapping of widget to laid out dimensions. If no such dimension
+  /// is provided, the widget will be skipped.
+  ///
+  /// Since widget sizes are calculated independently from the rest of the
+  /// paragraph, the [textScaleFactor] is ignored.
+  @override
+  void build(ui.ParagraphBuilder builder, { double textScaleFactor = 1.0, List<PlaceholderDimensions> dimensions }) {
+    assert(debugAssertIsValid());
+
+    final bool hasStyle = style != null;
+    if (hasStyle)
+      builder.pushStyle(style.getTextStyle(textScaleFactor: textScaleFactor));
+    if (dimensions != null && builder.placeholderCount < dimensions.length) {
+      PlaceholderDimensions currentDimensions = dimensions[builder.placeholderCount];
+      builder.addPlaceholder(
+        currentDimensions.size.width,
+        currentDimensions.size.height,
+        alignment,
+        baseline: currentDimensions.baseline,
+        baselineOffset: currentDimensions.baselineOffset,
+      );
+    }
+    if (hasStyle)
+      builder.pop();
+  }
+
+  /// Calls visitor on this [WidgetSpan]. There are no children spans to walk.
+  bool visitChildren(bool visitor(InlineSpan span)) {
+    if (!visitor(this))
+      return false;
+    if (children != null) {
+      for (InlineSpan child in children) {
+        if (!child.visitChildren(visitor))
+          return false;
+      }
+    }
+    return true;
+  }
```

Comment:
the documentation and the code don't match here

Reply:
(metapoint: when editing the code, once you've made the edit, always reread all nearby comments and documentation lines to verify that they are still useful, correct, and up to date.)